src/validator.cpp

```diff
@@ -902,11 +884,12 @@ void Validator::ValidatorImpl::validateMath(const std::string &input, const Comp
     // Iterate through ci/cn elements and remove cellml units attributes.
     XmlNodePtr mathNode = node;
     validateAndCleanMathCiCnNodes(node, component, variableNames, bvarNames);
-    // Get the MathML string (with cellml:units attributes already removed) and remove the CellML namespace.
-    // While the removeSubstring() approach for removing the cellml namespace before validating with the MathML DTD
-    // is not ideal, libxml does not appear to have a better way to remove a namespace declaration from the tree.
+
+    // Remove the cellml namespace definition.
+    mathNode->removeNamespaceDefinition(CELLML_2_0_NS);
```

Comment:
Here we are doing a proper libXml2 removal of the CellML namespace from the subdocument.